baker/GrapherBaker.tsx

```diff
@@ -82,123 +75,32 @@ export const renderDataPageOrGrapherPage = async (
-            />
-        )
-
-    // Enrich the datapage JSON with all the published charts this variable is
-    // being used in (aka "related charts") if none have been defined manually.
-    // We also want to exclude from the list the chart that is displayed at the
-    // top of the page to avoid the unnecessary redundancy.
-    datapageJson.allCharts =
-        datapageJson.allCharts ??
-        (await getRelatedChartsForVariable(id, [grapher.id]))
-
-    // Compliment the text-only content from the JSON with rich text from the
-    // companion gdoc, if any
-    const datapageGdoc = await getDatapageGdoc(
-        datapageJson.googleDocEditLink,
-        false,
-        publishedExplorersBySlug
-    )
-
-    // A datapage gdoc is optional. However, if a gdoc is referenced in the
-    // JSON, we assume the gdoc is expected to be used. Additionally, we need
-    // the datapage gdoc to be published so that images in the gdoc content are
-    // baked in bakeGdriveImages() (see also apiRouter.put("/gdocs/:id")). We
-    // also check that the datapage gdoc is of type "fragment" so that it
-    // doesn't get a URL when baked. If any of these conditions is not met, we
-    // log an error and fall back to a grapher page.
-    const dataPageNotPublishedError = `The data page for variable ${datapageJson.title} (${id}) has not been published.`
-    if (!datapageGdoc) {
-        logErrorAndMaybeSendToBugsnag(
-            new JsonError(
-                `${dataPageNotPublishedError} Please check that the googleDocEditLink ${datapageJson.googleDocEditLink} is correct.`
-            )
-        )
-        return renderGrapherPage(grapher)
-    }
+    // If we have a single Y variable and that one has a schema version >= 2,
+    // meaning it has the metadata to render a datapage, then we show the datapage
+    // based on this information. Otherwise we see if there is a legacy datapage.json
+    // available and if all else fails we render a classic Grapher page.
+    const yVariableIds = grapher
+        .dimensions!.filter((d) => d.property === DimensionProperty.y)
+        .map((d) => d.variableId)
+    if (yVariableIds.length === 1) {
+        const variableId = yVariableIds[0]
+        const variableMetadata = await getVariableMetadata(variableId)
 
-    if (
-        !datapageGdoc?.published ||
-        datapageGdoc.content.type !== OwidGdocType.Fragment
-    ) {
-        logErrorAndMaybeSendToBugsnag(
-            new JsonError(
-                `${dataPageNotPublishedError} Please check that ${ADMIN_BASE_URL}/admin/gdocs/${datapageGdoc.id}/preview has been published as a fragment.`
-            )
-        )
-        return renderGrapherPage(grapher)
+        if (
+            variableMetadata.schemaVersion !== undefined &&
+            variableMetadata.schemaVersion >= 2
+        ) {
+            return await renderDataPageV2({
+                variableId,
+                variableMetadata,
+                isPreviewing: true,
```

Comment:
Is this right?

Reply:
No the code as is works as intended - this is in the function that shows the preview so we do want to set isPreviewing to true here